feasto_core_clean_arch/storages/storage_implementation.py

```diff
@@ -0,0 +1,344 @@
+from ib_users.models import UserAccount
+from oauth2_provider.models import Application
+from oauth2_provider.models import AccessToken, RefreshToken
+from oauthlib.common import generate_token
+
+from feasto_core_clean_arch.models import OrderItem, Order
+from feasto_core_clean_arch.interactors.storage_interfaces.storage_interface import StorageInterface, ItemDTO, \
+    RestaurantDTO, UserDTO, OrderItemDTO, OrderDTO, LoginDTO
+from ..constants.enum import StatusType
+from ..exceptions.custom_exceptions import InvalidRestaurantOwnerId, InvalidRestaurantId, InvalidItemId, InvalidUserId
+from ..models import User,Item
+from typing import List
+
+from ..models import Restaurant
+
+
+class StorageImplementation(StorageInterface):
+    def get_user(self, user_id: str) -> UserDTO:
+        user = User.objects.get(id=user_id)
+        user_dto = UserDTO(
+            id= user.id,
+            name= user.name,
+            phone_no= user.phone_no
+        )
+        return user_dto
+
+    def validate_item_id(self, item_id:int):
+        item = Item.objects.filter(id=item_id).exists()
+        if not item:
+            raise InvalidItemId
+
+    def validate_restaurant_owner(self,restaurant_id:int, user_id:str):
+        rest = Restaurant.objects.get(id=restaurant_id)
+        if rest.user_id != user_id:
+            raise InvalidRestaurantOwnerId
+    def validate_user_id(self, user_id:str):
+        user = User.objects.filter(id=user_id).exists()
+        if not user:
+            raise InvalidUserId
+
+
+    def validate_restaurant_id(self, restaurant_id:int):
+        rest = Restaurant.objects.filter(id=restaurant_id).exists()
+        if not rest:
+            raise InvalidRestaurantId
+
+    def get_item(self, item_id: int, restaurant_id: int) -> ItemDTO:
+        item = Item.objects.get(id=item_id)
+
+        item_dto = ItemDTO(
+            name=item.name,
+            id=item.id,
+            available_quantity=item.available_quantity,
+            restaurant_id=item.restaurant_id
+        )
+        return item_dto
+
+    def add_item(self,
+                    restaurant_id: int, name: str = None, available_quantity: int = None) -> ItemDTO:
+        item = Item.objects.create(
+            name=name,
+            available_quantity=available_quantity,
+            restaurant_id=restaurant_id,
+        )
+
+        item_dto = ItemDTO(
+            name= item.name,
+            id= item.id,
+            available_quantity= item.available_quantity,
+            restaurant_id= item.restaurant_id
+        )
+
+        return item_dto
+
+    def add_restaurant(self,
+                 name: str, user_id: str, status: StatusType, location: str) -> RestaurantDTO:
+        user = User.objects.get(id = user_id)
+        restaurant = Restaurant.objects.create(
+            name=name,
+            user=user,
+            location=location,
+            status=status
+        )
+
+        user_dto = self.get_user(user_id)
```

Comment:
overfetching user details, can only fetch name
and we better to make this call in interactor